### DISCUSSION TITLE
[質問]: Value of type 'Results<RealmData>' has no member 'remove' というエラーが出てしまう

### DISCUSSION THREAD

**Discussion by sheep1857** [lit-iphone-course/lit-iphone-course#150] [Question-original] [answered]
### エラーの概要

削除機能を実装しようとしたところ、Value of type 'Results<RealmData>' has no member 'remove' というエラーが出てしまいます。

### GitHubのURL

https://github.com/sheep1857/GoshuinApp.git

### TODOリスト

- [X] GitHubに最新のコードをアップロードしました。
- [X] できるだけ具体的なタイトルを設定しました。

**Comment by txxxxc** [comment, marked as answer]
まずエラーが出る理由はResults型にremoveという関数がないからです！(多分配列型のremoveを使いたかったってことかな?)

解決策としては以下のような感じで、Realmから取得したデータを配列で保持すると良いと思います！
配列型にはremoveという関数があるので削除できるかと！！
```swift
var goshuinList: Results<RealmData>!
↓
var goshuinList: RealmData = []
```

また、RealmのResults型から配列への変換は以下のように行なえます！！
```swift
for object in realm.objects(RealmData.self) {
    goshuinList.append(object)
}

```
